### ISSUE TITLE
Remove API key dependencies by reverse engineer backend api's

### ISSUE THREAD

**Issue by dikkietrom** [Cormanz/smartgpt#36]
for google use curl
for openAI check https://github.com/FlamingFury00/Free-AUTO-GPT-with-NO-API
or through nat.dev https://nat.dev 's backend api
wolfram "
news "

I can add more info when it is not clear how to go about it.

**Comment by philip06** [comment]
I wouldn't remove API key dependencies for anything existing, APIs are always more reliable and faster. Also this wouldn't save you from configuration. Instead of an API key it'd be a cookie which you would have to keep changing or login credentials.

However, I think adding an integration to nat.dev is worth looking into to open up GPT-4 access.

**Comment by Cormanz** [comment]
> However, I think adding an integration to nat.dev is worth looking into to open up GPT-4 access.

I don't think this is a good idea unfortunately, `nat.dev` doesn't promote API use (although they technically don't prohibit it), and working with an unstable, and very-prone-to-rapid-changes API wouldn't be a great idea.

**Comment by dikkietrom** [comment]
at this point I cannot use the api because i have no access, the goal is not perse to use nat.dev but to simply be able to use gpt4 for automations. OpenAI has an open backend api, it has no documentation though but it can be copied from the networkview. It gives back clean text, the formatting is post fetch. I would not mind giving it a shot myself and make a separate plugin if you can give me some direction.

**Comment by philip06** [comment]
I looked at the ChatGPT backend, it's using an Authorization requestion header with a JWT. Meaning you might have trouble with the 25 message limit when testing. Something to consider.

**Comment by dikkietrom** [comment]
I wrote some stuff but I better consider this is your personal project that is meant to work for you and you only, and for reasons unknown you make it a github. At this point I cannot do anything with it but I will keep an eye on it to see where it is going.

**Comment by Cormanz** [comment]
> I wrote some stuff but I better consider this is your personal project that is meant to work for you and you only, and for reasons unknown you make it a github. At this point I cannot do anything with it but I will keep an eye on it to see where it is going.

The project is still solely maintained by me (although a lot of others have been helping a lot at this point), but it is a public project now and it should work for as many as we can.

Feel free to make a draft pull request for the code you wrote, I'll check it out.
